src/util/debug.cpp

```diff
@@ -0,0 +1,668 @@
+/*
+ * Copyright (C) 2011-2020 Daniel Scharrer
+ *
+ * This software is provided 'as-is', without any express or implied
+ * warranty.  In no event will the author(s) be held liable for any damages
+ * arising from the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose,
+ * including commercial applications, and to alter it and redistribute it
+ * freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not
+ *    claim that you wrote the original software. If you use this software
+ *    in a product, an acknowledgment in the product documentation would be
+ *    appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be
+ *    misrepresented as being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ */
+
+#include "util/debug.hpp"
+
+#include <ctime>
+#include <iostream>
+
+#include <boost/foreach.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/range/size.hpp>
+
+#include "loader/offsets.hpp"
+
+#include "setup/component.hpp"
+#include "setup/data.hpp"
+#include "setup/delete.hpp"
+#include "setup/directory.hpp"
+#include "setup/file.hpp"
+#include "setup/header.hpp"
+#include "setup/icon.hpp"
+#include "setup/info.hpp"
+#include "setup/ini.hpp"
+#include "setup/item.hpp"
+#include "setup/language.hpp"
+#include "setup/message.hpp"
+#include "setup/permission.hpp"
+#include "setup/registry.hpp"
+#include "setup/run.hpp"
+#include "setup/task.hpp"
+#include "setup/type.hpp"
+#include "setup/version.hpp"
+
+#include "stream/block.hpp"
+
+#include "util/fstream.hpp"
+#include "util/load.hpp"
+#include "util/log.hpp"
+#include "util/output.hpp"
+#include "util/time.hpp"
+
+#ifdef WASM_BUILD
```

Comment:
@Michal-Ston-Mobica see if you are fine with such changes

Reply:
I guess it's fine.